### DISCUSSION TITLE
Can not work for update the CTkProgressBar value, why?

### DISCUSSION THREAD

**Discussion by Barry-sze** [TomSchimansky/CustomTkinter#394] [General]
How to update a progress bar in a loop?

def update_progressbar():
       for n in range(10):
            progressbar.set(n/10)
     
progressbar = customtkinter.CTkProgressBar(master=frame_1,)
progressbar.place(x=20,y=380)
progressbar.set(0.0)

**Comment by TomSchimansky** [comment]
Your example is not working because the for loop happens instantly with no delay. For timed events there is the after() method in tkinter/customtkinter. You need to use the after() method, so that the UI is not blocked like if it would be if you use time.sleep() or something similar.
```python
import customtkinter


def start_progressbar(value):
    progressbar.set(value)

    if value < 1:
        # after() arguments: time in ms, function to call, arguments to the function
        app.after(1000, start_progressbar, value + 1/10)


app = customtkinter.CTk()

progressbar = customtkinter.CTkProgressBar(master=app)
progressbar.pack(padx=20, pady=20)
progressbar.set(0.0)

start_progressbar(0)

app.mainloop()
```
![Bildschirmfoto 2022-08-17 um 11 09 54](https://user-images.githubusercontent.com/66446067/185080983-e92a0cfa-e787-4da7-9e28-ee65665d03ad.png)

**Reply by Barry-sze** [reply]
How to use customtkinter progress_var like below:

from tkinter import *
from tkinter import ttk
import time
gui=Tk()
gui.title('Delftstack')
gui.geometry('600x400')

def StartProgress():
    for i in range(5):
        progress_var['value'] += 20
        stringvar.set(progress_var['value'])
        # get the value
        label.config(text=stringvar.get()+' %')
        # update value
        gui.update_idletasks()
        time.sleep(1)

def StopProgress():
    # stop progress
    progress_var.stop()
    # set zero percent
    stringvar.set('00.0 %')
    label.config(text=stringvar.get())

# create an instance of progress bar
progress_var=ttk.Progressbar(gui,orient=HORIZONTAL,length=400,mode='determinate')
progress_var.place(x=30,y=20)
start_btn=Button(gui,text='Next',command=StartProgress)
start_btn.place(x=30,y=50)
# string variable
stringvar=StringVar()
stringvar.set('00.0 %')
cancel_btn=Button(gui,text='Cancel',command=StopProgress)
cancel_btn.place(x=400,y=50)
label=Label(gui,text=stringvar.get(),font=('',12))
label.place(x=430,y=25)
gui.mainloop()

**Reply by TomSchimansky** [reply]
You cannot use `sleep` in your main GUI thread because it completely freezes the whole window!!!

**Reply by Barry-sze** [reply]
How to use app.after at for x in the range(10):
Print(x)
App.after(????,????)

**Comment by Barry-sze** [comment]
[https://www.delftstack.com/img/Tkinter/Text%20in%20Tkinter%20Progress%20Bar.gif](url)
